File: workspace/recommend/model/trade.py
from sqlalchemy import Column, Integer, String, BigInteger, BINARY, TIMESTAMP
from sqlalchemy.orm import declarative_base

Base = declarative_base()

class Trade(Base):
    __tablename__ = 'trade'
    trade_id = Column(BigInteger, primary_key=True)
    user_id = Column(BigInteger)
    category_id = Column(BigInteger)
    address_id = Column(Integer)
    title = Column(String(50))
    price = Column(Integer)
    trade_explain = Column(String)
    wish_count = Column(Integer)
    view_count = Column(Integer)
    trade_status = Column(Integer)
    created_date = Column(TIMESTAMP)
    update_date = Column(TIMESTAMP)
    thumb_nail_image = Column(String)
    buyer_id = Column(BigInteger)

File: workspace/backend/mpick/src/main/java/com/k9c202/mpick/trade/entity/Trade.java
package com.k9c202.mpick.trade.entity;

import com.k9c202.mpick.user.entity.Address;
import com.k9c202.mpick.user.entity.User;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import javax.persistence.*;
import java.sql.Timestamp;

import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Getter
@SuperBuilder
@NoArgsConstructor
@Entity
@Table(name = "trade")
public class Trade {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "trade_id")
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "category_id")
    private Category category;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "address_id")
    private Address address;

    @Column
    private String title;

    @Column
    private Integer price;

    @Column
    private String tradeExplain;

    @Column
    private Long wishCount;

    @Column
    private Long viewCount;

    @Column
    @Enumerated(EnumType.ORDINAL)
    private TradeStatus tradeStatus;

    @CreationTimestamp
    @Column(name = "created_date")
    private Timestamp tradeCreateDate;

    @UpdateTimestamp
    @Column(name = "update_date")
    private Timestamp tradeUpdateDate;

    public void increaseViewCount() { this.viewCount++; }

    public void increaseWishCount() { this.wishCount++; }

    public void decreaseWishCount() { this.wishCount--; }

    public void updateTrade(Category category, Address address, String title, Integer price, String tradeExplain, TradeStatus tradeStatus) {
        this.category = category;
        this.address = address;
        this.title = title;
        this.price = price;
        this.tradeExplain = tradeExplain;
        this.tradeStatus = tradeStatus;
    }
}
